py_gtfs_rt_ingestion/lib/s3_utils.py

```diff
@@ -64,6 +66,9 @@ def file_list_from_s3(bucket_name: str, file_prefix: str) -> List[str]:
             for obj in page["Contents"]:
                 filepaths.append(os.path.join("s3://", bucket_name, obj["Key"]))
 
+            if len(filepaths) > max_list_size:
+                break
+
```

Comment:
break if `max_list_size` hit. This will be checked on each new page which are ~2,000 files each. So might overshoot this value by that number, but shouldn't really matter for these purposes.